lib/internal/util/inspect.js

```diff
@@ -751,7 +770,8 @@ function formatValue(ctx, value, recurseTimes, typedArray) {
   if (ctx.seen.includes(value)) {
     let index = 1;
     if (ctx.circular === undefined) {
-      ctx.circular = new Map([[value, index]]);
+      ctx.circular = new SafeMap();
```

Comment:
Is `ctx` exposed to userland? We should not use a `SafeMap` if it's the case.

Reply:
I don't think it is – it is created in `inspect`, and functions using it are not exported to user-land.

https://github.com/nodejs/node/blob/d146b25928191cb4438ce5bf51ffce1856c1d171/lib/internal/util/inspect.js#L257-L260